### PR TITLE
HL-144 & HL-145 & HL-31 | Validation, Refactoring Styles and Typings

### PR DESCRIPTION
**Pull request by Nizar-Rahme**
## Description :sparkles:
- Made the typings of steps fields more strict. Each step can access its own fields; otherwise, it will throw an error.
- Got rid of most of the styled containers. 
- Implemented `$Grid` & `$GridCell` components that will function as wrappers to almost all the other elements and apply consistent styles across the application.
- Added validation to the second step of the application.
- As a bonus, the application now looks acceptable on mobile devices.

### CODE REVIEW

File: frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts [hunk 1]
```diff
@@ -35,9 +42,62 @@ type ExtendedComponentProps = {
   getDefaultSelectValue: (fieldName: keyof Application) => OptionType;
 };
 
+const ssnValidationTable = [
```
Comment (melniiv) [inline frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts:45]: https://www.npmjs.com/package/finnish-ssn
Could we use it?

File: frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts [hunk 2]
```diff
@@ -35,9 +42,62 @@ type ExtendedComponentProps = {
...
+
+  if (!validateSsn(number, checkCharacter)) return false;
+
+  return true;
+};
+
 const useApplicationFormStep2 = (
   application: Application
-): ExtendedComponentProps => {
+): UseApplicationFormStep2Props => {
```
Comment (melniiv) [inline frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts:100]: What is the point of renaming to "Use..."?

**Review by melniiv — APPROVED** [review]
looks good!

File: frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts (hunk 1 above)
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts:45]: I replaced our implementation with this dependency.

File: frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts (hunk 2 above)
Comment (Nizar-Rahme) [reply frontend/benefit/applicant/src/components/applications/forms/application/step2/useApplicationFormStep2.ts:100]: Nothing really 😅 
I saw some other hooks following this naming convention; so I thought, why not this one as well?

**Pull request merged**